### ISSUE TITLE
SonarScanner failing in pull requests

### ISSUE THREAD

**Issue by ajtribick** [CelestiaProject/Celestia#1268]
Example from https://github.com/CelestiaProject/Celestia/runs/4651908003?check_suite_focus=true :

```
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 51.821s
ERROR: Error during SonarScanner execution
INFO: Final Memory: 19M/67M
ERROR: You're not authorized to run analysis. No sonar.login or SONAR_TOKEN env variable was set
INFO: ------------------------------------------------------------------------
ERROR: 
ERROR: Re-run SonarScanner using the -X switch to enable full debug logging.
Error: Process completed with exit code 2.
```

**Comment by 375gnu** [comment]
Hi @ajtribick, secrets are not propagated to workflows if they are created from forks. Please use the main repository instead.

@levinli303 FYI

**Comment by ajtribick** [comment]
That's unfortunate, as it makes the contribution experience for developers outside the CelestiaProject organisation rather unfriendly. If you're unable to raise your PR from within the main repo, your PR is going to be marked as failing no matter what. This is not ideal for attracting new contributors!

I would therefore lean towards disabling this step in PRs until we have a better way to do this.

**Comment by ajtribick** [comment]
Maybe we want to consider using something like [bors](https://bors.tech) to run the CI? As I understand it, this merges to a staging area to run the CI, which would avoid needing to propagate secrets to forks.

**Comment by 375gnu** [comment]
Alternative can be deploying sonar to arm64 runner. This is a VM in oracle cloud so we can install everything (okay not everything as it runs on oraclelinux8) we need and keep secrets locally as well.

**Comment by 375gnu** [comment]
fixed with #1281
